### PR TITLE
Feature - QuocDoan - Video trending UI

### PR DESCRIPTION
**Pull request by at-quocdoan**
UI video trending
Dummy data for video trending
Page control video trending
simulator screen shot - iphone 8 - 2019-02-26 at 15 16 23

### CODE REVIEW

**Comment by at-lamle** [comment]
:+1: There are no linter violations.

File: MyApp/Model/Schema/Video.swift [hunk 1]
```diff
@@ -40,5 +40,7 @@ enum SectionType: Int {
 }
 
 final class Video {
-    #warning("Code in the future")
```
Comment (at-lamle) [inline MyApp/Model/Schema/Video.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/Model/Schema/Video.swift [hunk 2]
```diff
@@ -40,5 +40,6 @@ enum SectionType: Int {
...
```
Comment (at-lamle) [inline MyApp/Model/Schema/Video.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/View/Controller/Home/Cell/Trending/TrendingCell.swift [hunk 3]
```diff
@@ -10,5 +10,83 @@ import UIKit
 import MVVM
 
 final class TrendingCell: TableCell {
-    #warning("Code in the future")
+
+    // MARK: - Identifier
+    struct Identifier {
```
Comment (at-lamle) [inline MyApp/View/Controller/Home/Cell/Trending/TrendingCell.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/View/Controller/Home/Cell/Trending/TrendingCell.swift [hunk 4]
```diff
@@ -10,5 +10,88 @@ import UIKit
...
```
Comment (at-lamle) [inline MyApp/View/Controller/Home/Cell/Trending/TrendingCell.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/View/Controller/Home/HomeViewController.swift [hunk 5]
```diff
@@ -13,10 +13,11 @@ final class HomeViewController: ViewController {
 
     // MARK: - Identifier
     private struct Identifier {
+
```
Comment (at-lamle) [inline MyApp/View/Controller/Home/HomeViewController.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/View/Controller/Home/HomeViewController.swift [hunk 6]
```diff
@@ -14,9 +14,9 @@ final class HomeViewController: ViewController {
     // MARK: - Identifier
     private struct Identifier {
         static let header = "HomeHeaderView"
-        static let trendingNib = "TrendingCell"
```
Comment (at-lamle) [inline MyApp/View/Controller/Home/HomeViewController.swift]: WARNING - Lines should not have trailing whitespace.

**Comment by at-lamle** [comment]
Total linter violations in pull request: 6
Serious: 0
Warnings: 6

**Comment by at-lamle** [comment]
:+1: There are no linter violations.

**Comment by at-lamle** [comment]
Total linter violations in pull request: 9
Serious: 0
Warnings: 9

**Comment by at-quocdoan** [comment]
@at-bienle Please review this PR again. Thanks!

File: MyApp/View/Controller/Home/Cell/Trending/Collection/TrendingCollectionViewModel.swift [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+import UIKit
+
+final class TrendingCollectionCellViewModel {
+
+    // MARK: - Properties
+    var thumbnailImage: UIImage?
+    var title = ""
+
+    init(video: Video) {
+        self.thumbnailImage = video.thumbnailImage
```
Comment (at-bienle) [inline MyApp/View/Controller/Home/Cell/Trending/Collection/TrendingCollectionViewModel.swift]: Không cần `self`, tương tự với dòng dưới.

File: MyApp/View/Controller/Home/Cell/Trending/TrendingCell.swift [hunk 8]
```diff
@@ -10,5 +10,83 @@ import UIKit
...
+
+    private func configureTrendingView() {
+        guard let viewModel = viewModel else { return }
+        collectionView.setupSlide(with: viewModel.getImageTrending())
+    }
+}
+
+// MARK: - Scrollview delegate
+extension TrendingCell: UIScrollViewDelegate {
+    func scrollViewDidEndDecelerating(_ scrollView: UIScrollView) {
```
Comment (at-bienle) [inline MyApp/View/Controller/Home/Cell/Trending/TrendingCell.swift]: Thêm một dòng trống phía trên anh

File: MyApp/View/Controller/Home/HomeViewModel.swift [hunk 9]
```diff
@@ -12,12 +12,40 @@ final class HomeViewModel {
 
     // MARK: Properties
     let sections: [SectionType] = [.trending, .karaoke, .vpop, .bolero, .springMusic, .goldMusic, .channel]
+    var data: [[Video]] = [[], [], [], [], [], []]
```
Comment (at-bienle) [inline MyApp/View/Controller/Home/HomeViewModel.swift]: datas

**Comment by at-lamle** [comment]
:+1: There are no linter violations.

**Pull request merged**